Sofa/Component/MechanicalLoad/src/sofa/component/mechanicalload/NodalLinearDampingForceField.inl

```diff
@@ -132,91 +136,98 @@ sofa::core::objectmodel::ComponentState NodalLinearDampingForceField<DataTypes>:
+                                                       DataVecDeriv& _f, const DataVecCoord& _x,
+                                                       const DataVecDeriv& _v)
 {
-    if(!this->isComponentStateValid())
-        return;
+    if (!this->isComponentStateValid()) return;
 
     SOFA_UNUSED(_x);
 
-    sofa::helper::WriteAccessor<DataVecDeriv> f(_f);
+    auto f = sofa::helper::getWriteAccessor(_f);
     const VecDeriv& v = _v.getValue();
 
     // Constant and isotropic damping
-    if(isConstantIsotropic)
+    if (isConstantIsotropic)
     {
         const Real singleCoefficient = d_dampingCoefficient.getValue();
 
-        for(std::size_t i = 0; i < v.size(); ++i)
+        std::size_t i = 0;
+        for (auto& fi : f)
         {
-            f[i] -= v[i] * singleCoefficient;
+            fi -= v[i++] * singleCoefficient;
         }
     }
     else
     {
         const auto coefs = sofa::helper::getReadAccessor(d_dampingCoefficientVector);
 
-        if( !coefs.empty() )
+        if (!coefs.empty())
         {
             const std::size_t nbDampingCoeff = coefs.size();
+            std::size_t i = 0;
 
-            for(std::size_t i = 0; i < v.size(); ++i)
+            for (auto& fi : f)
             {
-                const auto coefId = std::min(i, nbDampingCoeff-1);
-                for(unsigned j = 0; j < Deriv::total_size; ++j)
+                const auto coefId = std::min(i, nbDampingCoeff - 1);
+                const auto& vi = v[i];
+                const auto& ci = coefs[coefId];
+                for (unsigned j = 0; j < Deriv::total_size; ++j)
                 {
-                    f[i][j] -= v[i][j] * coefs[coefId][j];
+                    fi[j] -= vi[j] * ci[j];
                 }
+                ++i;
             }
         }
     }
 }
 
-template<class DataTypes>
-void NodalLinearDampingForceField<DataTypes>::addDForce(const core::MechanicalParams* mparams, DataVecDeriv& d_df, const DataVecDeriv& d_dx)
+template <class DataTypes>
+void NodalLinearDampingForceField<DataTypes>::addDForce(const core::MechanicalParams* mparams,
+                                                        DataVecDeriv& d_df,
+                                                        const DataVecDeriv& d_dx)
 {
-    if(!this->isComponentStateValid())
-        return;
-    
+    if (!this->isComponentStateValid()) return;
+
     sofa::helper::WriteAccessor<DataVecDeriv> df(d_df);
     const VecDeriv& dx = d_dx.getValue();
 
     const Real bfactor = (Real)mparams->bFactor();
 
     // Constant and isotropic damping
-    if(isConstantIsotropic)
+    if (isConstantIsotropic)
     {
         const Real singleCoefficient = d_dampingCoefficient.getValue();
         Real factor = singleCoefficient * bfactor;
 
-        for (std::size_t i = 0; i < dx.size(); i++)
+        std::size_t i = 0;
+        for (auto& dfi : df)
         {
-            df[i] -= dx[i] * factor;
+            dfi -= dx[i++] * factor;
```

Comment:
I preferred the previous syntax. Here it makes things stranges. Generally, if you need the incremented index more than in one place, you should keep it. Because now, we must take care of incrementing ourself, so we keep the index + we don't treat each vector the same

Reply:
I fully agree